src/Microsoft.AspNetCore.Routing.Abstractions/Address.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public class Address
```

Comment:
Future design note:

I think it's fine for the first cut if address is simple and tightly coupled to the idea of a string identifier. We discussed the idea of address being polymorphic since MVC's addresses are a list of key-value-pairs

Reply:
Filed: https://github.com/aspnet/Routing/issues/564